### PR TITLE
add errors config

### LINKED ISSUE #2: Errors
It'd be nice if this util could capture errors

```js
{
  title: 'test',
  code: 'code',
  // one of
  error: true,
  error: 'Expected Error Message',
  error: /test on error/,
  error: SyntaxError,
  error: err => /foo/.test(err) && err instanceof SyntaxError,
},
```'

Or:

```js
{
  title: 'test',
  code: 'code',
  snapshot: true,
  error: true,
},
```

### PR DESCRIPTION
**Pull request by thejameskyle**
Resolves #2

### CODE REVIEW

File: src/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -20,6 +20,10 @@ exports[`throws an invariant if the plugin name is not inferable 1`] = `"The \`p
 
 exports[`throws error if fixture provided and code changes 1`] = `"Expected output to not change, but it did"`;
 
+exports[`throws error when error expected but no error thrown 1`] = `"undefined undefined undefined"`;
```
Comment (kentcdodds) [inline src/__snapshots__/index.test.js.snap]: Should this be a bunch of `undefined`s?

File: src/index.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ import * as recast from 'recast'
 import * as babel from 'babel-core'
 import stripIndent from 'strip-indent'
 import {oneLine} from 'common-tags'
+import {isError} from 'util'
```
Comment (kentcdodds) [inline src/index.js]: Looks like this is no longer needed.

**Comment by thejameskyle** [comment]
Can you change the complexity rules on eslint. The changes I have to make to the tester function makes the code more complicated to follow.

**Comment by kentcdodds** [comment]
Yeah, we can do that. Could you do it with:

```
// eslint-disable-next-line complexity
```

Right above the `tester` declaration. I'd prefer to discourage other functions get more complex.

**Comment by kentcdodds** [comment]
Cool! Thanks! I'm just about ready to merge. Could you explain what the `undefined`s are all about? Perhaps we can change it to be something more useful?

**Comment by kentcdodds** [comment]
If you want to avoid lint errors in travis, you could run `npm start validate` which will run the tests and linting. You can also add an `.opt-in` file to your root directory with the contents:

```
commit-msg
pre-commit
```

Which will run the validate script on commit.

**Comment by kentcdodds** [comment]
Hey James, I'm working on a bit of a refactor right now, so I think I'll update your PR and get it merged so I can rebase my stuff onto yours 👍 thanks for your help!

**Pull request merged**